src/zip.h

```diff
@@ -12,6 +12,8 @@
 #ifndef ZIP_H
 #define ZIP_H
 
+#include <zip/zip_export.h>
```

Comment:
But this breaks the basic, simple use case where users just _copy-paste_ files and compile them into the project, or if I want to just quickly compile one test, e.g.: `gcc -Wall -I../src test_write.c`.

This is super tiny library and in many use cases it's just vendored. 
Build system here is just a handy tool which let us compile and test the project on CI.

We can try to add `ZIP_EXPORT` macros to the `zip.h` file

Reply:
I was thinking that you tried to add this behavior in an old PR but failed to properly use it, since `generate_export_header` was already there, but the resulting header file was unused. I can add the macro instead.

Reply:
I don't remember exactly. Either it was some merge commit, or some copy-paste configuration for windows.

Reply:
I've added the macro directly in zip.h. Users can still pick the zip.c & zip.h files in their project, the default is empty ZIP_EXPORT.

For consumers compiling with CMake, even they should define ZIP_SHARED at consume time if zip is shared, but they can omit it, it doesn't hurt even on Windows (because zip.h doesn't export global symbols).